src/athena.hpp

```diff
@@ -69,6 +69,10 @@ struct LogicalLocation { // aggregate and POD type
   }
 };
 
+/// Defines the maximum size of the static array used in the IndexShape objects
+constexpr int NDIM = 3;
+static_assert(NDIM >= 3 &&
```

Comment:
I approve of the `static assert` here. But maybe this should be in the `domain.hpp` header next to `MeshShape`?

Reply:
Both parts or just the static assert?

Reply:
Just the static assert. My thinking was that the `NDIM` could be set to 
```C++
constexpr int NDIM = @NDIM@
```
and be set using the same `cmake` mechanism we use to set other compile-time parameters.